### PR TITLE
ADOPTERS: replace Documentation/production-users.md, add Chef

### PR DESCRIPTION
**Pull request by srenatus**
Let's get that list up to date -- for one thing, it's interesting, and it will probably help with #1379.

### CODE REVIEW

File: ADOPTERS.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Adopters
+
+This is a list of production adopters of Dex (in alphabetical order):
+
+- [Banzai Cloud](https://banzaicloud.com) is using Dex for authenticating to its Pipeline control plane and also to authenticate users against provisioned Kubernetes clusters (via Kubernetes OIDC support).
+- [Chef](https://chef.io) uses Dex for authenticating users in [Chef Automate](https://automate.chef.io/). The code is Open Source, available at [`github.com/chef/automate`](https://github.com/chef/automate).
```
Comment (JoelSpeed) [inline ADOPTERS.md]:
```suggestion
- [Chef](https://chef.io) uses Dex for authenticating users in [Chef Automate](https://automate.chef.io/). The code is Open Source, available at [`github.com/chef/automate`](https://github.com/chef/automate).
- [Pusher](https://pusher.com) uses Dex for authenticating users across their Kubernetes infrastructure (using Kubernetes OIDC support) in conjunction with the [OAuth2 Proxy](https://github.com/pusher/oauth2_proxy) for protecting web UIs.
```

**Comment by srenatus** [comment]
💭 I wonder what a good way to ping other folks would be... we've got a pretty silent google group... 🤔

**Comment by JoelSpeed** [comment]
I'd be tempted to tweet about it, I know a bunch of my followers use it so maybe they'll be keen to add their own? 🤷‍♂

**Comment by srenatus** [comment]
@JoelSpeed Why not? Worth a try for sure :+1:

**Pull request merged**